webviz_subsurface/plugins/_prod_misfit/_callbacks.py

```diff
@@ -0,0 +1,353 @@
+        Input(get_uuid(LayoutElements.HEATMAP_WELL_COLLECTIONS), "value"),
+        Input(get_uuid(LayoutElements.HEATMAP_WELL_COMBINE_TYPE), "value"),
+        Input(get_uuid(LayoutElements.HEATMAP_REALIZATIONS), "value"),
+        Input(get_uuid(LayoutElements.HEATMAP_FILTER_LARGEST), "value"),
+        Input(get_uuid(LayoutElements.HEATMAP_FIGHEIGHT), "value"),
+        Input(get_uuid(LayoutElements.HEATMAP_SCALE_COL_RANGE), "value"),
+        # prevent_initial_call=True,
+    )
+    def _update_heatmap_graph(
+        ensemble_names: List[str],
+        selector_dates: list,
+        selector_phases: list,
+        selector_well_names: list,
+        selector_well_collection_names: list,
+        selector_well_combine_type: str,
+        selector_realizations: list,
+        selector_filter_largest: int,
+        selector_figheight: int,
+        selector_scale_col_range: float,
+    ) -> List[wcc.Graph]:
+
+        well_names = _get_well_names_combined(
+            well_collections,
+            selector_well_collection_names,
+            selector_well_names,
+            selector_well_combine_type,
+        )
+
+        # if plot_type in ["diffplot", "rel_diffplot"]:
+        plot_type = "diffplot"
+        relative_diff = plot_type == "rel_diffplot"
+        dframe = makedf.get_df_diff_stat(
+            makedf.get_df_diff(
+                makedf.get_df_smry(
+                    input_provider_set,
+                    ensemble_names,
+                    ens_vectors,
+                    ens_realizations,
+                    selector_realizations,
+                    well_names,
+                    selector_phases,
+                    selector_dates,
+                ),
+                relative_diff=relative_diff,
+            )
+        )
+
+        # dframe = self.df_diff_stat.copy()
+
+        # # --- apply date filter
+        # dframe = dframe.loc[dframe["DATE"].isin(dates)]
+
+        # # --- apply ensemble filter
+        # dframe = dframe.loc[dframe["ENSEMBLE"].isin(ensemble_names)]
+
+        # # --- apply well filter
+        # dframe = dframe.loc[dframe["WELL"].isin(well_names)]
+
+        figures = makefigs.heatmap_plot(
+            dframe,
+            selector_phases,
+            vector_type="well",
+            filter_largest=selector_filter_largest,
+            figheight=selector_figheight,
+            scale_col_range=selector_scale_col_range,
+        )
+        return figures
+
+
+# ----------------------
+# --- help functions ---
+# ----------------------
+
+
+def _get_well_names_combined(
+    well_collections: Dict[str, List[str]],
+    selected_collection_names: list,
+    selected_wells: list,
+    combine_type: str = "intersection",
+) -> List[str]:
+    """Return union or intersection of well list and well collection lists."""
+
+    selected_collection_wells = []
+    for collection_name in selected_collection_names:
+        selected_collection_wells.extend(well_collections[collection_name])
```

Comment:
maybe like this? 
`selected_collection_wells = list(set([well_collections[name] for name in selected_collection_names])) `